homework-submission/homework_week2/index.js

```diff
@@ -0,0 +1,85 @@
+class TodoItems extends React.Component {
+  delete(key) {
+    this.props.delete(key);
+  }
+  createTasks = item => {
+    return (
+      <li onClick={() => this.delete(item.id)} key={item.id}>
+        {item.description} {item.deadline}
+      </li>
+    );
+  };
+  render() {
+    const todoEntries = this.props.entries;
+    const listItems = todoEntries.map(this.createTasks);
+    return <ul className="theList">{listItems}</ul>;
+  }
+}
+const Button = ({ text, clickHandler }) => {
+  return <button onClick={clickHandler}>{text}</button>;
+};
+
+class TodoList extends React.Component {
+  state = {
+    items: [
+      {
+        id: 1,
+        description: 'Get out of bed',
+        deadline: '2017-09-11',
+        done: true,
+      },
+      {
+        id: 2,
+        description: 'Brush teeth',
+        deadline: '2017-09-10',
+        done: false,
+      },
+      {
+        id: 3,
+        description: 'Eat breakfast',
+        deadline: '2017-09-09',
+        done: false,
+      },
+    ],
+  };
+  addItem = event => {
+    event.preventDefault();
+    if (event.target.value !== '') {
+      const newItem = {
+        description: event.target.inputElement.value,
+        id: Math.random(),
```

Comment:
maybe you can have a sequential ids rather than a random number
(hint: you can generate the next id by looking how many todos are already there)

Reply:
I'll try that, thanks!